### ISSUE TITLE
GitLab webhook fails in exception

### ISSUE THREAD

**Issue by slvwolf** [olipo186/Git-Auto-Deploy#36]
Seems GitLab integration is not working correctly. Getting following error from webhook:

```python
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 33991)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
    method()
  File "GitAutoDeploy.py", line 117, in do_POST
    repo_urls = self.get_repo_urls_from_request()
  File "GitAutoDeploy.py", line 169, in get_repo_urls_from_request
    elif user_agent.lower().find('bitbucket') != -1:
AttributeError: 'NoneType' object has no attribute 'lower'
----------------------------------------
```

Seems the header detection is not functioning correctly as changing the method to following works fine:

```python
    def get_repo_urls_from_request(self):
        """Parses the incoming request and extracts all possible URLs to the repository in question. Since repos can
        have both ssh://, git:// and https:// URIs, and we don't know which of them is specified in the config, we need
        to collect and compare them all."""
        import json
        #content_type = self.headers.getheader('content-type')
        length = int(self.headers.getheader('content-length'))
        body = self.rfile.read(length)
        data = json.loads(body)
        repo_urls = []
        if not 'repository' in data:
            print "ERROR - Unable to recognize data format"
            return repo_urls
        for k in ['url', 'git_http_url', 'git_ssh_url', 'push_data']:
            if k in data['repository']:
                repo_urls.append(data['repository'][k])
        return repo_urls
```

**Comment by olipo186** [comment]
What version of GitLab are you running?

**Comment by slvwolf** [comment]
* GitLab 7.9.2
* GitLab Shell 2.6.0
* GitLab API v3
* Ruby 2.1.5p273
* Rails 4.1.9

**Comment by olipo186** [comment]
As you mention it's probably the header detection that fails. When I wrote that part of the code, I testet against GitLab 7.13.5. Maybe the X-GitLab-Event header was introduces in a version of GitLab later than 7.9.2.

I modified the detection mechanism to include a additional condition that has been used earlier against older versions of GitLab. I haven't been able to test this properly though since I don't have access to ant older GitLab instances. Let's hope it'll do the job.

If you have the time to test it out, or debug the issue further, it would be greatly appreciated!
